Fixed generating single value properties from DTS in legacy

ollenorelius opened the pull request:
After slack discussion with @ulfalizer we believe the legacy DTS include generation has a bug where single value properties in the device tree are not generated properly, resulting in an internal python error. 

Something puts the single value as a scalar where it needs to be an array. This fix simply checks for that case and puts the scalar into an array if needed.

Example of non-working property in dts:

```
pwmleds {
	compatible = "pwm-leds";
	pwmled0: pwmled0 {
		pwms = <&pwm0>;
		label = "pwm-led-1";
		channel= <11>;
	};
};
```

This makes DTS generation throw this python error: 
```
Traceback (most recent call last):

  File "C:/Users/Olle/Documents/ncs/ncs/zephyr/scripts/dts/extract_dts_includes.py", line 546, in <module>
    main()
  File "C:/Users/Olle/Documents/ncs/ncs/zephyr/scripts/dts/extract_dts_includes.py", line 532, in main
    generate_defines()
  File "C:/Users/Olle/Documents/ncs/ncs/zephyr/scripts/dts/extract_dts_includes.py", line 455, in generate_defines
    generate_node_defines(node_path)
  File "C:/Users/Olle/Documents/ncs/ncs/zephyr/scripts/dts/extract_dts_includes.py", line 138, in generate_node_defines
    generate_prop_defines(node_path, prop)
  File "C:/Users/Olle/Documents/ncs/ncs/zephyr/scripts/dts/extract_dts_includes.py", line 93, in generate_prop_defines
    def_label, generic, deprecate=True)
  File "C:\Users\Olle\Documents\ncs\ncs\zephyr\scripts\dts\extract\globals.py", line 411, in extract_controller
    prop_array = build_cell_array(prop_values)
  File "C:\Users\Olle\Documents\ncs\ncs\zephyr\scripts\dts\extract\globals.py", line 336, in build_cell_array
    while index < len(prop_array):
TypeError: object of type 'int' has no len()
CMake Error at C:/Users/Olle/Documents/ncs/ncs/zephyr/cmake/dts.cmake:203 (message):
  command failed with return code: 1
Call Stack (most recent call first):
  C:/Users/Olle/Documents/ncs/ncs/zephyr/cmake/app/boilerplate.cmake:532 (include)

  CMakeLists.txt:8 (include)
```

ulfalizer commented on scripts/dts/extract/globals.py:
```diff
@@ -333,6 +333,11 @@ def build_cell_array(prop_array):
 
+    if isinstance(prop_array, int):
+    # Work around old code generating an integer for e.g.
+    # 'pwms = <&foo>'
```
Wrong indentation for comment. Looks fine otherwise.

Thanks!

ulfalizer replied:
Could you squash the indentation fix so it's just a single commit too?

Can `git push --force` after that.

ulfalizer replied:
Could make the commit title something like

    scripts: dts: Fix generation of single-value phandle/value array

to make CI happy.

ollenorelius committed:
scripts: dts: Fix generation of single-value phandle/value array

Put a type check in the offending method.
Now any scalar will be put in a list for
compatibility with following code.

Signed-off-by: Olle Norelius <norelius.olle@gmail.com>

ulfalizer commented:
Put a

    Signed-off-by: Olle Norelius <norelius.olle@gmail.com>

at the bottom of the commit message as well to make it go through CI. It's a legal thingy.

ulfalizer commented:
@ollenorelius 
Hmm... think you accidentally overwrote the old commit message when squashing commits.

Just says "Fixed indent" now. :)

ollenorelius commented:
I'm trying to abide by the Gitlint check, but I can't find out what it's mad about...

ulfalizer commented:
The bot edits the last message instead of posting a new one.

Here's the problem:

```
4: UC4 Line exceeds max length (75>72): "Now any scalar will be put in a list for compatibility with following code."
```

ollenorelius commented:
Man you are a saint
